api/client/contextdialer.go

```diff
@@ -52,6 +52,18 @@ func NewDirectDialer(keepAlivePeriod, dialTimeout time.Duration) ContextDialer {
 	}
 }
 
+// NewDialer makes a new dialer that connects to an Auth server either directly or via an HTTP proxy, depending
+// on the environment.
+func NewDialer(keepAlivePeriod, dialTimeout time.Duration) ContextDialer {
```

Comment:
What is the distinction, now, between NewDialer and NewDirectDialer? Should NewDialer be favored?

Could we make NewDirectDialer private? If not, please document the distinction in both method's godocs.